### DISCUSSION TITLE
Support for basic enums (not just backed one)

### DISCUSSION THREAD

**Discussion by chojnicki** [laravel/framework#40389] [Ideas]
I described it with details and gave examples in issue that was closed: https://github.com/laravel/framework/issues/40380

To summary: Laravel does not support basic enums, just backed ones. It is not specified anywhere in documentation, just crashes app while trying to cast one. In my opinion there is no reason to not support basic enums. Since Laravel officially supports Enums, basic one (that are described in first chapter of PHP enum documentation) should be implemented in first place before backed ones, not in other way around ;)

 If we do not need integer or string enumeration, we are forced to write ugly enums like this:

```
    case AAA = 'AAA';
    case BBB = 'BBB';
    case CCC = 'CCC';
```

instead just

```
    case AAA;
    case BBB;
    case CCC;
```

**Comment by ryangjchandler** [comment]
@chojnicki This isn't a clean way of doing things. All pure enums have a value of `null`, so the only way to obtain a variant/instance would be resolving by name which is odd imo.

**Comment by henzeb** [comment]
They are probably never going to support it. Try my package https://github.com/henzeb/enumhancer. It will allow you to use basic enums without too much hassle.

**Reply by henzeb** [reply]
Meanwhile, I finally extended the package with a way to use basic enums inside your controllers using implicit binding. So now on all places where they are 'forcing' you to use string-backed enums, you can now use basic-enums 

see: https://github.com/henzeb/enumhancer/blob/main/docs/binding.md
